sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java

```diff
@@ -148,28 +140,11 @@
                 .map(this::convertToLogQueryBatchResult);
     }
 
-    private String buildPreferHeaderString(LogsQueryOptions query) {
-        StringBuilder sb = new StringBuilder();
-        if (query.isIncludeRendering()) {
-            sb.append("include-render=true");
+    private Context updateContext(Duration serverTimeout, Context context) {
```

Comment:
>updateContext [](http://example.com/codeflow?start=20&length=13)

Isn't context a reference type? Do we need to return it?

Reply:
No, context is like a linked list. When new data is added, the head location changes and we need to use the new head reference. So, we have to return the context with updated reference. See [example](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java#L146)